evm/validation.py

```diff
@@ -16,7 +16,7 @@ def validate_is_bytes(value):
 
 
 def validate_is_integer(value):
-    if not isinstance(value, int):
+    if not type(value) is int:
```

Comment:
`if isinstance(value, int) and not isinstance(value, bool)` would break the core tests